src/App.tsx

```diff
@@ -15,49 +19,110 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE = 'none',
+  ALPHABET = 'alphabet',
+  LENGTH = 'length',
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+): string[] {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((curGood, nextGood) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return curGood.localeCompare(nextGood);
+
+      case SortType.LENGTH:
+        return curGood.length - nextGood.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, changeIsReversed] = useState(false);
+  const [sortType, changeSortType] = useState(SortType.NONE);
+
+  const sortByType = (typeOfSort: SortType) => {
+    changeSortType(typeOfSort);
+  };
```

Comment:
let's create functions for each type or just use changeSortType cuz now sortByType just a needless wrapper